### PR TITLE
Bluetooth: Controller: Fix use of pre-programmed PPI and cleanup includes

### PR DESCRIPTION
**Pull request by cvinayak**
Fix to not use pre-programmed PPI when NRF_TIMER0 is not
used as the radio event timer.

Remove redundant header file includes, introduce radio used
resources header file an cleanup includes. This was needed to
avoid redefinitions.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Use the timer instance ID, not NRF_TIMERx directly, so that it can be checked
+ * in radio_nrf5_ppi.h by the preprocessor.
+ */
+#if !defined(CONFIG_BT_CTLR_TIFS_HW)
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h:10]: Suggest to reverse order here, i.e. do 

```c
#if defined(CONFIG_BT_CTLR_TIFS_HW)
...
#else
...
#endif /* CONFIG_BT_CTLR_TIFS_HW */
```

as IMO it's nicer to have the true branch first

**Review by Thalley — COMMENTED** [review]
I don't know enough about the changes to approve, but generally from a generic code standpoint it seems fine and nice to get some more common functionality and less code duplication

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h]: done

**Review by aescolar — APPROVED** [review]
+1

**Review by carlescufi — APPROVED** [review]

**Pull request merged**